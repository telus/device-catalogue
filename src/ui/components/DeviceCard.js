import React from "react";
import FlexGrid from "@tds/core-flex-grid";
import Heading from "@tds/core-heading";
import Card from "@tds/core-card";
import PriceLockup from "@tds/core-price-lockup";
import Text from "@tds/core-text";
import Image from "@tds/core-image";
import Button from "@tds/core-button";
import ChevronLink from "@tds/core-chevron-link";
import Box from "@tds/core-box";
import DimpleDivider from "@tds/core-dimple-divider";

export default function DeviceCardItem(props) {
  return (
    <div className="device-card_item">
      <FlexGrid outsideGutter={false}>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={9}>
            <Card fullHeight variant="defaultWithBorder">
              <Box between={4}>
                <Image src={props.url} width={200} height={100} alt="device" />
                <Heading level="h3">{props.brand.toUpperCase()}</Heading>
                <Heading level="h2" tag="h3">
                  {props.name}
                </Heading>
                <PriceLockup
                  size="medium"
                  price={props.price}
                  bottomText="TELUS Easy Payment and Bring-It-Back applied. Read legal footnote. Taxes due upfront."
                  signDirection="left"
                />
                <br />
                <Text size="medium">Retail price: $470</Text>
                {/* <Button>Add to cart</Button>
                <ChevronLink href="#">Learn more</ChevronLink> */}
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
      <DimpleDivider />
    </div>
  );
}
