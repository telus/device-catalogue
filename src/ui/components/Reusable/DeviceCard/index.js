import React from 'react';
import Heading from '@tds/core-heading';
import Image from '@tds/core-image';
import Card from '@tds/core-card';
import Box from '@tds/core-box';
import Link from '@tds/core-link';
import Paragraph from '@tds/core-paragraph';
import Strong from '@tds/core-strong';
import Text from '@tds/core-text';
import HairlineDivider from '@tds/core-hairline-divider';
import { FootnoteLink } from '@tds/core-terms-and-conditions';
import PriceLockup from '@tds/core-price-lockup';
const DeviceCard = ({ img, path, text, brand, name, price}) => {
  const handleFootnoteClick = () => {
    console.log('footnote is clicked, terms and condition should open up from the bottom of the window');
  }
  return (
    <div className="device-card">
      <Card fullHeight>
        <Box between={4}>
          {path
           ? <Link href={path}>
              <div className="device-image">
                <Image 
                  src={img} 
                  width={200}
                  height={200}
                  alt="device-image"
                />
                <br/>
                {text}
              </div>
            </Link>
            :<div className="device-image">
              <Image 
                src={img} 
                width={200}
                height={200}
                alt="device-image"
              />
            </div>
          }
          {price &&
          <React.Fragment>
            <Strong id="device-brand">{brand}</Strong>
            <Heading level="h4" id="device-name" >{name}</Heading>
            <PriceLockup
              size="small"
              price={price}
              signDirection="left"
            />
            <HairlineDivider/>
            <Paragraph size="small">
              <Text>
                TELUS Easy Payment and Bring-It-Back applied.
                <FootnoteLink number={4} onClick={handleFootnoteClick} copy='en'/>
                <br/>
                Taxes due upfront.
              </Text>
            </Paragraph>
          </React.Fragment>
          }
        </Box>
      </Card>
    </div>
  )
}
export default DeviceCard;