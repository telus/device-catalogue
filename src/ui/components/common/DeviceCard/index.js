import React from "react";
import PropTypes from "prop-types";
import Card from "@tds/core-card";
import Box from "@tds/core-box";
import Heading from "@tds/core-heading";
import Image from "@tds/core-image";
import PriceLockup from "@tds/core-price-lockup";
import HairlineDivider from "@tds/core-hairline-divider";
import Paragraph from "@tds/core-paragraph";
import styled from "styled-components";
import { colorTelusPurple } from "@tds/core-colours";

const DeviceNameContainer = styled(Box)`
  align-items: flex-start;
`;

const DeviceName = styled.span`
  font-weight: 400;
  color: ${colorTelusPurple};
  font-size: 1.3rem;
  line-height: 1.29;
  letter-spacing: -0.8px;
`;

const DeviceCard = ({ device, image, imageAlt = "Device image" }) => {
  if (!device) {
    return null;
  }

  return (
    <Box vertical={3}>
      <Card variant="defaultWithBorder" spacing="narrow">
        <Box between={2}>
          <Box horizontal={5}>
            <Image src={image} alt={imageAlt} width={200} height={300} />
          </Box>
          <DeviceNameContainer between={1}>
            <Heading level="h3" tag="span">
              {device.brand}
            </Heading>
            <DeviceName>{device.name}</DeviceName>
          </DeviceNameContainer>
          <PriceLockup
            size="small"
            price={device.price.toString()}
            signDirection="left"
          />
        </Box>
        <Box vertical={4} between={4}>
          <HairlineDivider />
          <Paragraph>
            TELUS Easy Payment and Bring-It-Back applied. Read legal footnote
            Taxes due upfront.
          </Paragraph>
        </Box>
      </Card>
    </Box>
  );
};

DeviceCard.propTypes = {
  device: PropTypes.object,
  image: PropTypes.any,
  imageAlt: PropTypes.string,
};

export default DeviceCard;
