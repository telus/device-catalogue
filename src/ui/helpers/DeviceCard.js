import React from "react";
import FlexGrid from "@tds/core-flex-grid";
import Heading from "@tds/core-heading";
import Box from "@tds/core-box";
import Image from "@tds/core-image";
import Card from "@tds/core-card";
import HairlineDivider from "@tds/core-hairline-divider";
import PriceLockup from "@tds/core-price-lockup";
import Paragraph from "@tds/core-paragraph";

/**
 * A component that renders a Telus device card with image, name and product info
 * @param {Object} device
 * @param {imagefile} image
 */
const DeviceCard = ({ device, image }) => {
  return (
    <Card>
      <FlexGrid.Row horizontalAlign="center">
        <Box vertical={4}>
          <Image
            src={image}
            alt={`Image of ${device.name}`}
            height={200}
            width={160}
          />
        </Box>
      </FlexGrid.Row>
      <Box vertical={3} between={1}>
        <Heading level="h3" tag="h2">
          {" "}
          {device.brand}
        </Heading>
        <Heading level="h2" tag="h3">
          {" "}
          {device.name}
        </Heading>
      </Box>
      <Box>
        <PriceLockup
          size="medium"
          price={`${device.price}`}
          signDirection="left"
        />
      </Box>
      <Box between={3} vertical={4}>
        <HairlineDivider />
        <Paragraph>
          TELUS Easy Payment and Bring-It-Back applied. Read legal footnote
          Taxes due upfront.{" "}
        </Paragraph>
      </Box>
    </Card>
  );
};
export default DeviceCard;
