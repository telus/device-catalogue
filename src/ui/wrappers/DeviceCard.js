import React from 'react';
import Heading from '@tds/core-heading';
import Box from '@tds/core-box';
import PriceLockup from '@tds/core-price-lockup';
import Paragraph from '@tds/core-paragraph';
import Image from '@tds/core-image';
import Card from '@tds/core-card';
import HairlineDivider from '@tds/core-hairline-divider';

const DeviceCard = ({ name, brand, price, image }) => (
  <Card variant="defaultWithBorder">
    <Image src={image} width={158} height={200} alt="iPhone" />
    <Box vertical={2} style={{ textAlign: 'left' }}>
      <Heading level="h3" tag="h2">
        {brand}
      </Heading>
      <Paragraph size="large">{name}</Paragraph>
    </Box>
    <PriceLockup size="small" price={price} signDirection="left" />
    <Box vertical={4}>
      <HairlineDivider />
    </Box>
    <Paragraph size="medium">
      TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes
      due upfront.
    </Paragraph>
  </Card>
);

export default DeviceCard;
