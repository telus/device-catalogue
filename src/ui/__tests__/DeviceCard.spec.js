import React from 'react';
import { render } from 'enzyme';
import DeviceCard from '../wrappers/DeviceCard';

describe('DeviceCard component', () => {
  it('user should see device image', () => {
    const image = 'device.jpg';
    const wrapper = render(<DeviceCard image={image} />);

    const imageSrc = wrapper.find('img').attr('src');

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('img').length).toEqual(1);
    expect(imageSrc).toBe(image);
  });

  it('user should see device information', () => {
    const wrapper = render(
      <DeviceCard name="P30 Lite" brand="HUAWEI" price="519.58" />
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.text()).toMatch(/P30 Lite/i);
    expect(wrapper.text()).toMatch(/HUAWEI/i);
    expect(wrapper.text()).toMatch(/519.58/i);
  });
});
