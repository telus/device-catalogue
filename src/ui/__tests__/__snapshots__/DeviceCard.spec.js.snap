// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceCard component user should see device image 1`] = `"<img src=\\"device.jpg\\" width=\\"158\\" height=\\"200\\" alt=\\"iPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"><div style=\\"text-align:left\\" class=\\"sc-bdvvtL hgLkkr\\"><h2 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\"></h2><p class=\\"Paragraph__StyledParagraph-efl81j-0 iuGNvs\\"></p></div><div class=\\"sc-bdvvtL iEjIjW sc-gsDKAQ UcpVv sc-gsDKAQ UcpVv\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-dkPtRN gPiEds sc-dkPtRN gPiEds\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"></span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-hKwDye fMvSxW dlaLOW\\"></span></span></div></div></div></div></div><div class=\\"sc-bdvvtL bBQzKm\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"></div><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p>"`;

exports[`DeviceCard component user should see device information 1`] = `"<img width=\\"158\\" height=\\"200\\" alt=\\"iPhone\\" class=\\"Image__StyledImage-b16p20-0 cNmOoM\\"><div style=\\"text-align:left\\" class=\\"sc-bdvvtL hgLkkr\\"><h2 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 kwSJnX\\">HUAWEI</h2><p class=\\"Paragraph__StyledParagraph-efl81j-0 iuGNvs\\">P30 Lite</p></div><div class=\\"sc-bdvvtL iEjIjW sc-gsDKAQ UcpVv sc-gsDKAQ UcpVv\\"><div class=\\"sc-bdvvtL ipXlaM\\"><div class=\\"PriceLockup__StyledRateTextWrapper-crmhv4-4 inieGs\\"><div class=\\"sc-bdvvtL chLtDX sc-dkPtRN gPiEds sc-dkPtRN gPiEds\\"><div class=\\"sc-bdvvtL chLtDX\\"><span aria-hidden=\\"false\\" data-testid=\\"dollarSign\\" class=\\"PriceLockup__StyledDollarSign-crmhv4-2 bsmnqF\\">$</span><span data-testid=\\"priceValue\\" aria-hidden=\\"false\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\">519.58</span><span data-testid=\\"priceValue\\" class=\\"PriceLockup__StyledPriceValue-crmhv4-1 pjRFu\\"><span class=\\"Text__StyledText-sc-1m0rr67-0 sc-hKwDye fMvSxW dlaLOW\\"></span></span></div></div></div></div></div><div class=\\"sc-bdvvtL bBQzKm\\"><hr class=\\"HairlineDivider__StyledHairlineDivider-biamlk-0 BvZCb\\"></div><p class=\\"Paragraph__StyledParagraph-efl81j-0 jukEfT\\">TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes due upfront.</p>"`;
